### ISSUE TITLE
Incompatibility with pandas 2.0

### ISSUE THREAD

**Issue by rob-p** [biocore-ntnu/pyranges#299]
Hi,

  First, thanks for the very useful package!  We use pyranges in our tool [`pyroe`](https://github.com/COMBINE-lab/pyroe).  Recently, a user reported [an issue](https://github.com/COMBINE-lab/simpleaf/issues/80) which we tracked down to pyranges.  Initially, it worked on our environment, but not under a clean virtual environment with freshly-installed packages.

  Ultimately, it seems the issue is related to pyranges making use of a deprecated keyword argument that became an actual error in pandas 2.

```
  File "/software/miniconda3/envs/simpleaf/lib/python3.10/site-packages/pyroe/make_txome.py", line 638, in make_splici_txome
    introns = gr.features.introns(by="transcript")
  File "/software/miniconda3/envs/simpleaf/lib/python3.10/site-packages/pyranges/genomicfeatures.py", line 254, in introns
    result = pyrange_apply(_introns2, by_gr, exons, **kwargs)
  File "/software/miniconda3/envs/simpleaf/lib/python3.10/site-packages/pyranges/multithreaded.py", line 293, in pyrange_apply
    result = call_f(function, nparams, df, odf, kwargs)
  File "/software/miniconda3/envs/simpleaf/lib/python3.10/site-packages/pyranges/multithreaded.py", line 23, in call_f
    return f.remote(df, odf, **kwargs)
  File "/software/miniconda3/envs/simpleaf/lib/python3.10/site-packages/pyranges/genomicfeatures.py", line 607, in _introns2
    introns.Feature.cat.add_categories(["intron"], inplace=True)
  File "/software/miniconda3/envs/simpleaf/lib/python3.10/site-packages/pandas/core/accessor.py", line 112, in f
    return self._delegate_method(name, *args, **kwargs)
  File "/software/miniconda3/envs/simpleaf/lib/python3.10/site-packages/pandas/core/arrays/categorical.py", line 2475, in _delegate_method
    res = method(*args, **kwargs)
TypeError: Categorical.add_categories() got an unexpected keyword argument 'inplace'
```

While we can, of course, gate on the major pandas version for the time being, it would be very useful to have this fixed upstream so that the package is compatible with the current major release of pandas (especially as it brings very substantial speed improvements).

Please let us know if you need any more info.

Thanks!
Rob

**Comment by endrebak** [comment]
I will look at this tomorrow.

**Comment by endrebak** [comment]
Should be fixed in 0.0.123, which is out now. If something is not as expected, feel free to reopen.

Thanks for reporting :D
